### PR TITLE
feat(docz-core): expose filtered env variables

### PR DESCRIPTION
**Pull request by lupuszr**
### Description

Add dotenv to support reading .env files
Expose env varables to browser which start with
REACT_APP_
ANGULAR_APP_
VUE_APP_
NODE_

### Review

- [ ] Check the copy
- [ ] ...
- [ ] ...

### Pre-merge checklist

- [ ] ...
- [ ] ...

### Screenshots

| Before | After |
| ------ | ----- |
| Image  | Image |

### CODE REVIEW

File: packages/docz-core/src/bundlers/webpack/dotenv.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+const NODE_ENV = process.env.NODE_ENV
+// if (!NODE_ENV) {
+//   throw new Error(
+//     'The NODE_ENV environment variable is required but was not specified.'
+//   )
+// }
+
+// https://github.com/bkeepers/dotenv#what-other-env-files-can-i-use
+const dotenvFiles = [
+  `$dotenv}.${NODE_ENV}.local`,
```
Comment (pedronauck) [inline packages/docz-core/src/bundlers/webpack/dotenv.ts]: Missing `{` here after `$`

File: packages/docz-core/src/bundlers/webpack/dotenv.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import fs from 'fs'
```
Comment (pedronauck) [inline packages/docz-core/src/bundlers/webpack/dotenv.ts]: please, move this file inside to `src/config` to be together with the `paths.ts` file...

**Review by pedronauck — CHANGES_REQUESTED** [review]

File: packages/docz-core/src/bundlers/webpack/dotenv.ts (hunk 2 above)
Comment (lupuszr) [reply packages/docz-core/src/bundlers/webpack/dotenv.ts]: 👍

File: packages/docz-core/src/bundlers/webpack/dotenv.ts (hunk 1 above)
Comment (lupuszr) [reply packages/docz-core/src/bundlers/webpack/dotenv.ts]: 😱 Sure

**Pull request merged**